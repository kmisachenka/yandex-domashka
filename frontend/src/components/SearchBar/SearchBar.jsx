import React from 'react';

import styles from './SearchBar.module.scss';

export default function SearchBar() {
  return (
    <form className={styles.form}>
      <label className={styles.label}>
        <input type="search" className={styles.input} placeholder="Поиск" />
      </label>
      <button type="button" className={styles.button}>
        Найти
      </button>
    </form>
  );
}
