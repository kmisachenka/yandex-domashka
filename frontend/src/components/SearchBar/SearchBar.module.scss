.form {
  display: flex;
}

.label {
  height: 32px;
  position: relative;
  width: 100%;

  &::after {
    background: url('./svg/clear.svg') no-repeat;
    content: '';
    height: 12px;
    position: absolute;
    right: 10px;
    top: 10px;
    width: 12px;
  }
}

.input {
  border: 2px solid #e6e6e6;
  border-radius: 4px 0 0 4px;
  box-sizing: border-box;
  font-family: Helvetica, serif;
  font-size: 15px;
  line-height: 28px;

  min-height: 28px;

  padding: 0 12px;

  position: relative;

  width: 100%;

  &:focus {
    border-color: #fc0;
    outline: none;
  }

  &::placeholder {
    color: #000;
    opacity: 0.2;
  }
}

.button {
  appearance: none;

  background: #e6e6e6;
  border: 0;
  border-radius: 0 4px 4px 0;

  color: #000;

  font-family: Helvetica, serif;
  font-size: 15px;

  height: 100%;
  line-height: 32px;
  margin: 0;

  padding: 0 16px;

  &:hover {
    background-color: #3ba6ff;
    color: #fff;
    transition: background-color 0.2s ease, color 0.1s ease-in;
  }

  &:active {
    background-color: #008000;
    color: #fff;
  }
}
