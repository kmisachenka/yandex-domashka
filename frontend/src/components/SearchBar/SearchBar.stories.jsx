import React from 'react';
import { storiesOf } from '@storybook/react';
import SearchBar from './SearchBar';

storiesOf('Поиск', module)
  .add('строка поиска', () => (
    <SearchBar />
  ));
