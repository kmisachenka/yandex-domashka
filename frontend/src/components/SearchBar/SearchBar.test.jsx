import React from 'react';
import { shallow } from 'enzyme';

import SearchBar from './SearchBar';

describe('SearchBar', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<SearchBar />);
    expect(wrapper).toMatchSnapshot();
  });
});
