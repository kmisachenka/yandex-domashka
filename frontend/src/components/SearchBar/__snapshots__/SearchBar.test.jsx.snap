// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should match the snapshot 1`] = `
<form
  className="form"
>
  <label
    className="label"
  >
    <input
      className="input"
      placeholder="Поиск"
      type="search"
    />
  </label>
  <button
    className="button"
    type="button"
  >
    Найти
  </button>
</form>
`;
